lambda/aws-bedrock-data-automation/create-blueprint/lambda.py

```
@@ -0,0 +1,170 @@
+    
+
+def get_schema(bucket_name: str, schema_key: str) -> Dict[str, Any]:
+    """
+    Get and parse schema JSON from S3 bucket
+    
+    Args:
+        bucket_name: Name of the S3 bucket containing the schema
+        schema_key: Object key (path) of the schema file in S3
+        
+    Returns:
+        Dict[str, Any]: Parsed schema content
+        
+    Raises:
+        ValueError: If schema file is not found or invalid JSON
+        ClientError: If S3 access fails
+    """
+    try:
+        logger.info("Retrieving schema from S3", extra={
+            "bucket": bucket_name,
+            "key": schema_key
+        })
+        s3_client = boto3.client('s3')
+        response = s3_client.get_object(
+            Bucket=bucket_name,
+            Key=schema_key
+        )
+        schema_content = json.loads(response['Body'].read().decode('utf-8'))
+        logger.info("Successfully retrieved and parsed schema", extra={
+            "schema_size": len(json.dumps(schema_content)),
+            "bucket": bucket_name,
+            "key": schema_key
+        })
+        
+        return schema_content
+    
+    except Exception as e:
+        logger.error("Unexpected error retrieving schema", extra={
+            "bucket": bucket_name,
+            "key": schema_key,
+            "error": str(e)
+        })
+        raise
+
+#@logger.inject_lambda_context
+def handler(event, context: LambdaContext):
+    """
+    Lambda handler function
+    """
+    try:
+        logger.info(f"Received event: {json.dumps(event)}")
+        # Get input bucket and key from environment variables
+       
+        if 'detail' not in event:
+            logger.error("Event detail section is missing")
+            return {
+                'statusCode': 400,
+                'body': json.dumps({
+                    'message': 'Invalid event format',
+                    'error': 'detail section is missing'
+                })
+            }
```

Looks like you're getting an event from Event Bridge here and want to enforce data parsing and validation using Parser utility. You can use [event_parser](https://docs.powertools.aws.dev/lambda/python/latest/utilities/parser/#event-parser) decorator and fail fast if the event that arrives in Lambda is not an EB event. Or you can use [parse](https://docs.powertools.aws.dev/lambda/python/latest/utilities/parser/#parse-function) function to have better control over exception mechanism.

It also provides a fully typed variable with auto-completion in the IDE.

A pseudo-code is something like this:

```python
from aws_lambda_powertools.utilities.parser import parse
from aws_lambda_powertools.utilities.parser.models import EventBridgeModel

def lambda_handler(event: dict, context):
    try:
        # Manually parse the incoming event into the custom model
        parsed_event: EventBridgeModel = parse(model=EventBridgeModel, event=event)

        return {"statusCode": 200, "body": f"Event from {parsed_event.source}, type: {parsed_event.detail_type}"}
    except ValidationError as e:
        return {"statusCode": 400, "body": f"Validation error: {str(e)}"}
```

---

Good suggestion. May implement it in next build . For now I have already added the  specific validations.